### ISSUE TITLE
Entries with future publish date are only published after clearing the Stache

### ISSUE THREAD

**Issue by FlorianMoser** [statamic/cms#7480]
### Bug description

The collection tag won't automatically list an entry, that isn't published instantly. Create an entry with a publish date in the past, and it will be shown. Create an entry with a publish date in the future and it will not be listed, even after the date expired.

This happens with Antlers as well as with Blade templates and it doesn't seem to be an issue with view caching. Using `php artisan view:clear` doesn't help. But `php please stache:clear` does the trick. So it seems to be an issue with the Stache.

### How to reproduce

1. Create a new Statamic project
2. Log into the control panel and create a new collection "Blog"
3. Edit the collection and enable "Enable Publish Dates"
4. Edit the "date" field in its blueprint and enable "Time Enabled", save
5. Edit the `home.antlers.html` file, add the collection tag:
```
<ul>
    {{ collection:blog }}
        <li>{{ title }}</li>
    {{ /collection:blog }}
</ul>
```
6. Create a Blog-entry with a date/time in the past and publish
7. Reload the frontend: you see the entry
8. Create a Blog-entry with a date a minute ahead of now and publish
9. Wait a minute and reload the Blog-entry list in the control panel. You see the entry should be published, it gets a green dot
10. Reload the frontend. The new entry doesn't appear

Only after `php please stache:clear` and reloading the frontend, the entry becomes visible.

### Environment

```yaml
Environment
Application Name: Statamic
Laravel Version: 9.50.2
PHP Version: 8.1.11
Composer Version: 2.4.2
Environment: local
Debug Mode: ENABLED
URL: statdate.test
Maintenance Mode: OFF

Cache
Config: NOT CACHED
Events: NOT CACHED
Routes: NOT CACHED
Views: CACHED

Drivers
Broadcasting: log
Cache: statamic
Database: mysql
Logs: stack / single
Mail: smtp
Queue: sync
Session: file

Statamic
Addons: 0
Antlers: runtime
Stache Watcher: Enabled
Static Caching: Disabled
Version: 3.4.2 Solo
```

### Installation

Fresh statamic/statamic site via CLI

### Antlers Parser

runtime (new)

**Comment by jasonvarga** [comment]
Duplicate of #2217
